src/clients/sync.rs

```diff
@@ -75,8 +78,14 @@ pub(crate) async fn sync_db(
 
             last_height = std::cmp::max(last_height, block_height);
 
-            for (ssk, vk, psk) in addresses.iter() {
+            for (i, (ssk, vk, psk)) in addresses.iter().enumerate() {
                 if vk.owns(&note) {
+                    if let Some(existing_addresses) = existing_addresses {
+                        if existing_addresses.get(i).is_none() {
+                            addresses_to_create.add_assign(1);
```

Comment:
This way, you are loosing the index of the address you need to create.

Reply:
Additionally:
- if you find multiple notes belonging to the same address, you're counting them multiple times
- if there are no existing_address you're not counting at all
- even if the above are fixed, what if I have 5 existing address and I find a note that belong to the address with index=50? Remember that we check notes against all MAX_Addresses


I suggest to change the behaviors to return the "max_address_index" found instead of the missing addresses count, and then create all the missing address to reach "max_address_index"


Anyway, some tests for this new feature would be welcome